frontend/src/app/attributes/EditIntents.tsx

```diff
@@ -1,14 +1,85 @@
-import React from 'react'
+import React, { useState, useEffect } from 'react'
 import './EditIntents.css'
+import { IntentData } from './Intent'
 
 interface EditIntentsProps {
-    // TODO: Fill out the props for your component
+    intents: Array<IntentData>
+    userIntentMap: { [code: string]: { [side: string]: boolean } }
+    updateIntent(code: string, side: string, value: boolean): void
+}
+
+const Subjects = ({ topic }) => {
+    return <h6>{topic}</h6>
+}
+
+const Seeking = ({ code, value, updateIntent }) => {
+    const [checked, setIsChecked] = useState(value)
+    const checkHandler = () => {
+        updateIntent(code, 'seeking', !checked)
+        setIsChecked(!checked)
+    }
+    useEffect(() => {
+        setIsChecked(value)
+    }, [value])
+
+    return (
+        <div>
+            <input type="checkbox" checked={checked} onChange={checkHandler} />
+            <span>I want help on this.</span>
+            <br />
+        </div>
+    )
+}
+
+const Giving = ({ code, value, updateIntent }) => {
+    const [checked, setIsChecked] = useState(value)
+    const checkHandler = () => {
+        updateIntent(code, 'seeking', checked)
+        setIsChecked(!checked)
+    }
+
+    return (
+        <div>
+            <input type="checkbox" checked={checked} onChange={checkHandler} />
+            <span>I can help others with this.</span>
+            <br />
+        </div>
+    )
+}
+
+function transformUserIntents(intent, userIntentMap) {
+    const { code, name } = intent
+    const isSeeking = userIntentMap?.[code]?.seeking || false
+    const isGiving = userIntentMap?.[code]?.giving || false
+    // Get seeking and giving from attributesMap
+    return { code, name, isSeeking, isGiving }
 }
 
 // TODO: Implement your component
 // eslint-disable-next-line no-empty-pattern
-const EditIntents = ({}: EditIntentsProps) => {
-    return <p> Marilyn Elaine Flowers was here! </p>
+const EditIntents = ({ intents = [], userIntentMap = {}, updateIntent }: EditIntentsProps) => {
+    const userIntents = intents.map((intent) => transformUserIntents(intent, userIntentMap))
+    console.log(userIntents)
+    return (
+        <div>
+            {userIntents.map((intent) => (
+                <div key={intent.code}>
+                    <Subjects topic={intent.name} />
+                    <Seeking
```

Comment:
The code for `Seeking` and `Giving` are very similar. Can you create a single component called `IntentCheckbox` and use it for both?

Here is what the `EditIntents` component should look like if you do that:

```jsx
<Subjects topic={intent.name} />
<IntentCheckbox
  side="seeking"
  code={intent.code}
  value={intent.isSeeking}
  updateIntent={updateIntent}
/>
<IntentCheckbox
  side="giving"
  code={intent.code}
  value={intent.isGiving}
  updateIntent={updateIntent}
/>
```

Notice how we use the same component twice, one for each side. And we add a new prop called `side` that will allow the component to use that variable instead of hard-coding giving or seeking.